CMakeLists.txt

```diff
@@ -101,6 +101,19 @@ if (KR_ENABLE_VELOC OR KR_ENABLE_HDF5_PARALLEL)
    target_link_libraries(resilience PRIVATE MPI::MPI_CXX)
 endif()
 
+option(KR_ENABLE_EXEC_SPACES "enable resilient execution spaces" OFF)
```

Comment:
Added this flag to gate all resilient exec space stuff

Reply:
This is minor, but would something like `cmake_dependent_option(KR_ENABLE_CUDA_EXEC_SPACE "enable resilient cuda execution space" ON "KR_ENABLE_EXEC_SPACES;_cuda_found GREATER -1" OFF)` work?